lamstare/experiments/plt_ood.py

```diff
@@ -1,112 +1,71 @@
-import logging
-from typing import Union
+from functools import lru_cache
 
-from matplotlib.axes import Axes
-from matplotlib.lines import Line2D
+import matplotlib.pyplot as plt
+import numpy as np
 import pandas
 import yaml
-from lamstare.experiments.plt_test import fetch_dptest_res
-from lamstare.infra import Record
+from matplotlib.axes import Axes
+from matplotlib.lines import Line2D
+from pandas import DataFrame
+
+from lamstare.experiments.plt_test import COLOR, fetch_dptest_res, parse_record_dict_to_df
 from lamstare.infra.ood_database import OODRecord
 from lamstare.utils.plot import sendimg
-from collections import defaultdict
-import numpy as np
-import matplotlib.pyplot as plt
-from pandas import DataFrame
 
-COLOR = [
-    "red",
-    "yellowgreen",
-    "dodgerblue",
-    "mediumpurple",
-    "orange",
-    "lightsalmon",
-    "hotpink",
-    "tan",
-    "cyan",
-    "navy",
-    "orchid",
-]
-
-with open("/mnt/data_nas/cc/LAMstare_new/lamstare/release/OOD_DATASET.yml", "r") as f:
+
+with open(
+    "/mnt/workspace/public/multitask/LAMstare/lamstare/release/new_OOD_DATASET.yml", "r"
+) as f:
     OOD_DATASET = yaml.load(f, Loader=yaml.FullLoader)
 OOD_DATASET = (
-    DataFrame(OOD_DATASET["OOD_TO_HEAD_MAP"])
-    .T.reset_index(names="Dataset")
-    .set_index("Dataset")
-    .infer_objects()
+    DataFrame(OOD_DATASET["OOD_TO_HEAD_MAP"]).T.rename_axis("Dataset").infer_objects()
 )
-# print(OOD_DATASET)
-# Dict[head, {filepath, weights, test_virial}]
+# normalize weights
+for index in OOD_DATASET.keys():
+    if "weight" in index:
+        efv_weight = OOD_DATASET[index]
+        efv_weight /= efv_weight.mean()
+print(OOD_DATASET)
 
 OOD_DATASET_STD = pandas.read_csv(
     "/mnt/workspace/cc/LAMstare_new/lamstare/release/ood_data_std.csv"
-)
-# Dataset, energy_std, force_std, virial_std
+).infer_objects()
 OOD_DATASET_STD.set_index("Dataset", inplace=True)
-OOD_DATASET_STD = OOD_DATASET_STD.infer_objects()
-
-
-def parse_record_dict_to_df(data: dict) -> DataFrame:
-    # Convert dictionary to a list of tuples
-    data_tuples = [
-        (dataset, training_steps, metrics)
-        for dataset, training_steps_dict in data.items()
-        for training_steps, metrics in training_steps_dict.items()
-    ]
-
-    # Extract the first metrics dictionary to get the keys
-    columns = ["Dataset", "Training Steps"] + list(data_tuples[0][2].keys())
-
-    # Create DataFrame
-    df = DataFrame(
-        [
-            (dataset, training_steps, *metrics.values())
-            for dataset, training_steps, metrics in data_tuples
-        ],
-        columns=columns,
-    )
-
-    # Set multi-index
-    df.set_index(["Dataset", "Training Steps"], inplace=True)
-    # sort by Dataset
-    df.sort_index(level=0, inplace=True)
-    return df
-
+print(OOD_DATASET_STD)
 
+@lru_cache
```

Comment:
I don't think we need to cache this. My guess is if we cache this, any updates on OODRecord data will be skipped.

Reply:
This only persists in one python interpreter session.